### PR TITLE
Remove padding in `DialogContent`

### PR DESCRIPTION
**Pull request by JoanaMMoreira**
Preview at https://14c812da.saltdesignsystem-storybook.pages.dev/?path=/story/core-dialog--long-content

### CODE REVIEW

**Comment by joshwooding** [comment]
I don't think this is the correct way to solve it. DialogContent should have padding

**Comment by joshwooding** [comment]
Before:
image
image

After:
image
image

Without scrolling looks fine, but once there's a scrollbar it looks a little tight? Might not be an issue though

**Comment by origami-z** [comment]
> Without scrolling looks fine, but once there's a scrollbar it looks a little tight? Might not be an issue though

We probably need to live with this, unless there's a way to add padding when scroll bar appears

**Comment by JoanaMMoreira** [comment]
> > Without scrolling looks fine, but once there's a scrollbar it looks a little tight? Might not be an issue though
> 
> We probably need to live with this, unless there's a way to add padding only when scroll bar appears

I don't think there's a nice way of adding padding for the scroll bar, there are some hacks like adding a transparent thicker border to it but it's not really ideal

**Review by origami-z — COMMENTED** [review]
Can we update QA stories, so that right spacing change could be captured in Chromatic

File: .changeset/bright-dogs-beg.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@salt-ds/core": patch
+---
+
+Fix `DialogContent` spacing
```
Comment (origami-z) [inline .changeset/bright-dogs-beg.md:5]: Maybe be a little more specific? So when user reads this, they could know where to look for the diff in their own app.

**Review by origami-z — APPROVED** [review]

**Comment by origami-z** [comment]
@navkaur76 does any of padding removal impact pref dialog pattern?

**Review by joshwooding — APPROVED** [review]

**Pull request merged**